Intermittent test failure WaterMarkedBackpressureIT

In main branch pipeline

```
Error:  Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 6.073 s <<< FAILURE! - in io.helidon.reactive.webserver.WaterMarkedBackpressureIT
Error:  io.helidon.reactive.webserver.WaterMarkedBackpressureIT.linear  Time elapsed: 5.024 s  <<< FAILURE!
java.lang.AssertionError: 

Expected: is <101L>
     but: was <143L>
	at io.helidon.reactive.webserver.WaterMarkedBackpressureIT.linear(WaterMarkedBackpressureIT.java:122)

[INFO] 
[INFO] Results:
[INFO] 
Error:  Failures: 
Error:    WaterMarkedBackpressureIT.linear:122 
```

[logs_3116.zip](https://github.com/helidon-io/helidon/files/11111198/logs_3116.zip)

And here https://github.com/helidon-io/helidon/actions/runs/4736275822/jobs/8423499776?pr=6644

```
[INFO] Running io.helidon.webserver.WaterMarkedBackpressureIT
Error:  Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 20.823 s <<< FAILURE! - in io.helidon.webserver.WaterMarkedBackpressureIT
Error:  io.helidon.webserver.WaterMarkedBackpressureIT.autoFlush  Time elapsed: 15.393 s  <<< FAILURE!
java.util.concurrent.CompletionException: java.util.concurrent.TimeoutException
	at io.helidon.webserver.WaterMarkedBackpressureIT.autoFlush(WaterMarkedBackpressureIT.java:175)
Caused by: java.util.concurrent.TimeoutException
	at io.helidon.webserver.WaterMarkedBackpressureIT.autoFlush(WaterMarkedBackpressureIT.java:175)
```

And now:
```
Running io.helidon.webserver.WaterMarkedBackpressureIT
Error:  Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 20.887 s <<< FAILURE! - in io.helidon.webserver.WaterMarkedBackpressureIT
Error:  io.helidon.webserver.WaterMarkedBackpressureIT.autoFlush  Time elapsed: 15.376 s  <<< FAILURE!
java.util.concurrent.CompletionException: java.util.concurrent.TimeoutException
	at io.helidon.webserver.WaterMarkedBackpressureIT.autoFlush(WaterMarkedBackpressureIT.java:175)
Caused by: java.util.concurrent.TimeoutException
	at io.helidon.webserver.WaterMarkedBackpressureIT.autoFlush(WaterMarkedBackpressureIT.java:175)

[INFO] 
[INFO] Results:
[INFO] 
Error:  Failures: 
Error:    WaterMarkedBackpressureIT.autoFlush:175 » Completion java.util.concurrent.Time...
[INFO] 
Error:  Tests run: 3, Failures: 1, Errors: 0, Skipped: 0
```

And here: https://github.com/helidon-io/helidon/actions/runs/4813688292/jobs/8570534396?pr=6699

```
Running io.helidon.webserver.WaterMarkedBackpressureIT
Error:  Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 20.927 s <<< FAILURE! - in io.helidon.webserver.WaterMarkedBackpressureIT
Error:  io.helidon.webserver.WaterMarkedBackpressureIT.autoFlush  Time elapsed: 15.402 s  <<< FAILURE!
java.util.concurrent.CompletionException: java.util.concurrent.TimeoutException
	at io.helidon.webserver.WaterMarkedBackpressureIT.autoFlush(WaterMarkedBackpressureIT.java:175)
Caused by: java.util.concurrent.TimeoutException
	at io.helidon.webserver.WaterMarkedBackpressureIT.autoFlush(WaterMarkedBackpressureIT.java:175)
```

--- a/webserver/webserver/src/test/java/io/helidon/webserver/WaterMarkedBackpressureIT.java
+++ b/webserver/webserver/src/test/java/io/helidon/webserver/WaterMarkedBackpressureIT.java
@@ -39,6 +39,7 @@ import org.testng.annotations.Test;
 import static io.helidon.webserver.BackpressureStrategy.AUTO_FLUSH;
 import static io.helidon.webserver.BackpressureStrategy.LINEAR;
 import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.greaterThan;
 import static org.hamcrest.Matchers.is;
 
 public class WaterMarkedBackpressureIT {
@@ -145,7 +146,10 @@ public class WaterMarkedBackpressureIT {
                                 .observeOn(executorService)
                                 // Never flush!
                                 // 5 bytes per chunk
-                                .map(l -> DataChunk.create(false, ByteBuffer.wrap((String.format("%05d", l)).getBytes())))
+                                .map(l -> String.format("%05d", l))
+                                .map(String::getBytes)
+                                .map(ByteBuffer::wrap)
+                                .map(bb -> DataChunk.create(false, bb))
                                 .onCompleteResumeWith(Single.never())
                         );
                     }))
@@ -165,7 +169,7 @@ public class WaterMarkedBackpressureIT {
                         receivedSize.addAndGet(bytes.length);
                         String data = new String(bytes);
                         chunk.release();
-                        return !data.equals("00101");
+                        return !data.equals("00100");
                     })
                     .ignoreElements()
                     .onErrorResumeWithSingle(t -> {
@@ -175,7 +179,8 @@ public class WaterMarkedBackpressureIT {
                     .await(TIMEOUT);
 
 
-            assertThat(receivedSize.get(), is(510L));
+            // Stochastic test as watermarking depends on Netty's flush callbacks
+            assertThat(receivedSize.get(), greaterThan(499L));
         } finally {
             if (webServer != null) {
                 webServer.shutdown().await(TIMEOUT);

